src/types/serialized-type.ts

```diff
@@ -1,6 +1,96 @@
 import { BytesList } from "../serdes/binary-serializer";
 const { bytesToHex, slice } = require("../utils/bytes-utils");
 
+/**
+ * The base class for all binary-codec types
+ */
+class SerializedTypeClass {
+  protected bytes: Buffer = Buffer.alloc(0);
+
+  /**
+   * Write the bytes representation of a SerializedType to a BytesList
+   *
+   * @param list The BytesList to write SerializedType bytes to
+   */
+  toBytesSink(list: BytesList): void {
+    list.put(this.bytes);
+  }
+
+  /**
+   * Get the hex representation of a SerializedType's bytes
+   *
+   * @returns hex String of this.bytes
+   */
+  toHex(): string {
+    return this.toBytes().toString("hex").toUpperCase();
+  }
+
+  /**
+   * Get the bytes representation of a SerializedType
+   *
+   * @returns A buffer of the bytes
+   */
+  toBytes(): Buffer {
+    if (this.bytes) {
+      return this.bytes;
+    }
+    const bl = new BytesList();
```

Comment:
nit: prefer to not use abbreviations since they're a bit hard to read without context, recommend 
```javascript
const bytesList = new BytesList();
```